docs/source/developers/components_exports.rst

```diff
@@ -0,0 +1,494 @@
+Furthermore, bundles are generated on the fly, and only the source files are
+required to be present in the source tree.
+
+Options related to static assets are used to specify the directory where the
+source files are kept, and enumerate the JavaScript and CSS bundles that your
+component needs.
+
+By default, the source assets are looked up in ``static`` directory within the
+package directory. To change this behavior, we can use the ``static_dir``
+option. ::
+
+    [exports]
+
+    static_dir = assets
+
+.. note::
+    The static directory is expected to contain 'js' and 'css' subdirectories.
+    There is currently no way to change this.
+
+Bundles are defined using two options::
+
+    [exports]
+
+    js_bundles =
+        article: autoscroll, comment
+        summary: autoscroll, imagebox
+
+    css_bundles =
+        article: article_layout
+        summary: summary_layout
+
+This configuration creates two JavaScript bundles, and two CSS bundles. The
+JavaScript bundle 'article' contains concatenated and minified sources of
+'static/js/autoscroll.js' and 'static/js/comment.js'. As you can see, the
+static assets directory and 'js/' subdirectory are automatically added to the
+name, as is the '.js' extension.
+
+For more information on working with static assets, see
+:doc:`working_with_static_assets`.
+
+.. warning::
+    Note that using bundle names that are defined in other components will
+    cause the bundles to be merged and the source assets from those bundles
+    will be concatenated in the order the components themselves are registered.
+
+Hooks
+-----
+
+Hooks are functions that are executed for events. In context of component
+exports, these events are system events emitted by the supervisor.
+
+Every function that is going to be used as a hook must be decorated with a
+:py:func:`~librarian.core.exports.hook` decorator. ::
+
+    from librarian.core.exports import hook
+
+    @hook('initialize')
+    def on_initialize(supervisor):
+        # do something when component is initializing...
+
+Any of the system and custom events can be used. For the full list of events
+and their meaning, see :doc:`list_of_system_events`.
+
+Of these, actually useful ones are probably 'initialize',
+'component_member_loaded', 'init_complete', and 'background'. The 'background'
+even is interesting in particular, as it allows the component to repeatedly
+execute code on an interval. More information on event handling and supervisor
+hooks can be found in :doc:`working_with_events`.
+
+Once we have the decorated functions, we can list them in the configuration
+file using the ``hooks`` option::
+
+    [exports]
+
+    hooks = hooks.on_initialize
+
+In this case, we have an ``on_initialize`` function in a ``hooks`` module.
+
+Plugins
+-------
+
+Plugins are classes and functions that follow the `Bottle's plugin API
+<http://bottlepy.org/docs/dev/plugindev.html>`_. Just like middleware, plugins
```

Comment:
Not sure if it makes a difference, but ideally this should link to the stable version of the doc at [Plugin API](http://bottlepy.org/docs/stable/plugindev.html)